app/src/test/java/org/oppia/android/app/activity/route/ActivityRouterModuleTest.kt

```diff
@@ -102,17 +109,36 @@ class ActivityRouterModuleTest {
   }
 
   @Test
-  fun testInjectActivityRouterModule_isInstanceOfActivityRouter() {
-    val activityRouter = retrieveActivityRouter()
-    assertThat(activityRouter).isInstanceOf(ActivityRouter::class.java)
+  fun testInjectActivityRouterModule_routeToRecentlyPlayedActivity() {
+    activityRule.scenario.onActivity { activity ->
+      assert(
+        destinationRoutes.containsKey(
+          DestinationScreen.DestinationScreenCase.RECENTLY_PLAYED_ACTIVITY_PARAMS
+        )
+      )
+      destinationRoutes[DestinationScreen.DestinationScreenCase.RECENTLY_PLAYED_ACTIVITY_PARAMS]
+        .let { route ->
+          val intent = route?.createIntent(
+            activity,
+            DestinationScreen.getDefaultInstance()
+          )
+          assertThat(intent).hasComponentClass(RecentlyPlayedActivity::class.java)
+        }
+    }
```

Comment:
``let`` isn't needed here and is actually complicating things. You can just do something like:

```kotlin
val route = destinationRoutes[RECENTLY_PLAYED_ACTIVITY_PARAMS]
val intent = route?.createIntent(context, DestinationScreen.getDefaultInstance())
assertThat(intent).hasComponentClass(RecentlyPlayedActivity::class.java)
```